### ISSUE TITLE
BUG:  When you click on a Trial session that is in a New status, the suffixes (S, SL, L, etc.)  are not displaying in the Eligible Case list

### ISSUE THREAD

**Issue by ttlenard** [ustaxcourt/ef-cms#8106]
**Describe the Bug**
This was reported by a court user.  She was working on trial sessions under the NEW tab, and noticed that the suffixes were missing from the docket numbers listed under the Eligible cases.

Note: If you click on a trial session from the Open or Closed tab, this issue does not appear to occur. It is only the trial sessions that are in a NEW status. 

**Business Impact/Reason for Severity**

**In which environment did you see this bug?**
MIG
TEST
PROD

**Who were you logged in as?**
Petitions Clerk

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**
Using the application

**To Reproduce**
1. Log in as a petitions clerk (I was on TEST for this scenario)
2. Click on Trial Sessions
3. Click on the New tab
4. Find a Small Session type (or any session type, but it is easiest to see missing suffixes from a small session type). I selected 2-7-22 Tampa Florida.
5. Click on the Location to open the Trial session
6. Notice that none of the docket numbers have the suffix at the end.  (You can easily see this with the CDP/Lein/Levy cases, they should have an "L" at the end. If you are in a small session, these case types should have an "SL" at the end, but they are missing from the docket # in the eligible case list.) 
7. Select a docket record from the eligible case list (I selected the first case from the Eligible case list - 104-20.), and notice that when the case loads, you can see the "S" or "SL" at the end of the docket # from the main case page.
8. Note: I found this trial session under the All tab and clicked on it from there, and the suffixes are missing when you access it this way as well.  Pretty sure it has to do with the case status of being NEW.

**Expected Behavior**
When viewing eligible cases in a trial session under the New tab, or if a trial session is in the NEW status, you should be able to see the Suffixes in the docket #.

**Actual Behavior**
A clear and concise description of what actually happened.
When viewing eligible cases in a trial session under the New tab, or if a trial session is in the NEW status, if the case has a suffix, the suffixes are missing from view.

**Screenshots**

Here is a small session that I clicked on in TEST. Notice that there is a CDP (Lein/Levy) case at the top of the list, but the "SL" is not showing up in the Docket No.  Also, since this is a small session, there should be an "S" in the suffix for all of the Eligible cases.
![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/651969ad-e0cd-4558-92c6-e29ad64ff1f0)

**Desktop (please complete the following information):**
 - Browser [e.g. chrome, safari] Edge

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Ready for Bugs(Created 10-4-21)
Definition used: A failure or flaw in the system which produces an incorrect or undesired result that deviates from the expected result or behavior. (Note: Expected results are use cases that have been documented in past user stories as acceptance criteria and test cases, and do not include strange behavior unrelated to use cases.)

The following criteria must be met in order for the development team to begin work on the bug.

The bug must:
 - Be focused on solving a user problem
 - Contain data for all fields in the bug template, so the team can pick it up and begin working immediately

Process: If the unexpected results are new use cases that have been identified, but not yet built, new acceptance criteria and test cases should be captured in a new user story and prioritized by the product owner. 

If the Court is not able to reproduce the bug, add the “Unable to reproduce” tag. This will provide visibility into the type of support that may be needed by the Court. In the event that the Court cannot reproduce the bug, the Court will work with Flexion to communicate what type of troubleshooting help may be needed.

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Bug fix has been validated in the Court's test environment

**Engineering**
 - [ ] Automated test scripts have been written
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [x] Deployed to the Court's test environment

**Comment by kkoskelin** [comment]
This was a matter of creating an attribute `docketNumberWithSuffix` on the `EligibleCase` entity.
This attribute was frequently something which was calculated using a formatter/helper on the client-side.  But it was constantly being calculated over and over again, and eventually I believe we refactored code so that it was calculated just once in the `Case` entity.  It was also being calculated in the `PublicCase` entity.
However, when we removed the calculation from the formatter, we neglected to account for the fact that the raw case being provided was actually in the shape of the `EligibleCase` entity, which did not have its own attribute `docketNumberWithSuffix`.

Adding it to this entity aligned it with `Case` and `PublicCase`, and the hyperlink will be fixed in a PR to the court's test environment. https://github.com/ustaxcourt/ef-cms/pull/1755
